sky/utils/kubernetes/generate_static_kubeconfig.sh

```diff
@@ -33,8 +33,10 @@ else
     exit 1
 fi
 
-echo "Creating the Kubernetes Service Account with minimal RBAC permissions."
-kubectl apply -f - <<EOF
+# Check if SKYPILOT_SA_NAME env var exists
+if [ -z ${SKYPILOT_SA_NAME+x} ]; then
```

Comment:
Can we elaborate the comment a bit? Why we do we skip the creation when `SKYPILOT_SA_NAME` does not exist?